lib/http_aws_sigv4.c

```diff
@@ -200,6 +200,15 @@ static CURLcode make_headers(struct Curl_easy *data,
   }
 
   for(l = data->set.headers; l; l = l->next) {
+    char *ptr = strchr(l->data, ':');
+    /* if the user header is in format 'name:' with no value then it is being
+       used to ignore an internal header so do not add it to the header list */
+    if(ptr) {
+      for(++ptr; ISSPACE(*ptr); ++ptr)
+        ;
```

Comment:
I think we can insist on `name:` with no spaces at all when removing headers, so this check could be simplified a little.